lib/pc_util.c

```diff
@@ -263,3 +288,123 @@ void pc_bounds_merge(PCBOUNDS *b1, const PCBOUNDS *b2)
 	if ( b2->ymax > b1->ymax ) b1->ymax = b2->ymax;
 }
 
+static uint32_t srid_mask = 0x20000000;
+static uint32_t m_mask = 0x40000000;
+static uint32_t z_mask = 0x80000000;
+
+uint8_t *
+pc_bounds_to_bounding_diagonal_wkb(
+	const PCBOUNDS *bounds, const PCSCHEMA *schema, size_t *wkbsize)
+{
+	uint8_t *wkb, *ptr;
+	uint32_t wkbtype;
+	size_t size;
+
+	wkbtype = 2; /* WKB LINESTRING */
+	size = 1 + 4 + 4 + (2 * 2 * 8); /* endian + type + npoints + 2 dbl pts */
+
+	if ( schema->srid != 0 )
+	{
+		wkbtype |= srid_mask;
+		size += 4;
+	}
+
+	wkb = pcalloc(size);
+	ptr = wkb;
+
+	ptr = wkb_set_char(ptr, machine_endian()); /* Endian flag */
+	ptr = wkb_set_uint32(ptr, wkbtype); /* Geometry type */
+
+	if ( schema->srid != 0 )
+	{
+		ptr = wkb_set_uint32(ptr, schema->srid); /* SRID */
+	}
+
+	ptr = wkb_set_uint32(ptr, 2); /* NPOINTS = 2 */
+
+	// point 1
+	ptr = wkb_set_double(ptr, bounds->xmin);
+	ptr = wkb_set_double(ptr, bounds->ymin);
+
+	// point 2
+	ptr = wkb_set_double(ptr, bounds->xmax);
+	ptr = wkb_set_double(ptr, bounds->ymax);
+
+	if ( wkbsize )
+		*wkbsize = size;
+
+	return wkb;
+}
+
+uint8_t *
+pc_bounds_xyzm_to_bounding_diagonal_wkb(
+	const PCSTATS *stats, const PCSCHEMA *schema, size_t *wkbsize)
+{
```

Comment:
Why not simply `pc_stats_to_bounding_diagonal_wkb(const PCSTATS *stats, size_t *wkbsize)` ?
`schema` could be retrieved as `stats->min->schema`